src/servers/graphql-old-server.ts

```diff
@@ -228,12 +229,24 @@ const resolvers = {
         return true
       },
     }),
-    noauthAddInvoice: async (_, { username, value }, { logger }) => {
-      const wallet = await getWalletFromUsername({ username, logger })
-      return wallet.addInvoice({ selfGenerated: false, value })
+    noauthAddInvoice: async (_, { username, value }) => {
+      const lnInvoice = await addInvoiceForRecipient({
+        recipient: username,
+        amount: value,
+      })
+      if (lnInvoice instanceof Error) throw lnInvoice
+      return lnInvoice.paymentRequest
```

Comment:
@samerbuna @dolcalmi is it ok to return a value here instead of a promise?

I just picked this up on rebase after #435 was merged

Reply:
Ah nevermind, resolves in the same way: https://stackoverflow.com/a/42750371